Doc/library/string.templatelib.rst

```diff
@@ -0,0 +1,313 @@
+
+       A tuple of all interpolated values in the template.
+
+       >>> name = "World"
+       >>> t"Hello {name}!".values
+       ('World',)
+
+       The ``values`` tuple is always the same length as the
+       ``interpolations`` tuple. It is equivalent to
+       ``tuple(i.value for i in template.interpolations)``.
+
+   .. describe:: iter(template)
+
+       Iterate over the template, yielding each string and
+       :class:`Interpolation` in order.
+
+       >>> name = "World"
+       >>> list(t"Hello {name}!")
+       ['Hello ', Interpolation('World', 'name', None, ''), '!']
+
+       Empty strings are *not* included in the iteration:
+
+       >>> name = "World"
+       >>> list(t"Hello {name}{name}")
+       ['Hello ', Interpolation('World', 'name', None, ''), Interpolation('World', 'name', None, '')]
+
+   .. describe:: template + other
+                 template += other
+
+       Concatenate this template with another, returning a new
+       :class:`!Template` instance:
+
+       >>> name = "World"
+       >>> list(t"Hello " + t"there {name}!")
+       ['Hello there ', Interpolation('World', 'name', None, ''), '!']
+
+       Concatenation between a :class:`!Template` and a ``str`` is *not* supported.
+       This is because it is ambiguous whether the string should be treated as
+       a static string or an interpolation. If you want to concatenate a
+       :class:`!Template` with a string, you should either wrap the string
+       directly in a :class:`!Template` (to treat it as a static string) or use
+       an :class:`!Interpolation` (to treat it as dynamic):
+
+       >>> from string.templatelib import Template, Interpolation
+       >>> template = t"Hello "
+       >>> # Treat "there " as a static string
+       >>> template += Template("there ")
+       >>> # Treat name as an interpolation
+       >>> name = "World"
+       >>> template += Template(Interpolation(name, "name"))
+       >>> list(template)
+       ['Hello there ', Interpolation('World', 'name', None, '')]
+
+
+.. class:: Interpolation(value, expression="", conversion=None, format_spec="")
+
+   Create a new :class:`!Interpolation` object.
+
+   :param value: The evaluated, in-scope result of the interpolation.
+   :type value: object
+
+   :param expression: The text of a valid Python expression, or an empty string.
+   :type expression: str
+
+   :param conversion: The optional :ref:`conversion <formatstrings>` to be used, one of r, s, and a.
+   :type conversion: ``Literal["a", "r", "s"] | None``
+
+   :param format_spec: An optional, arbitrary string used as the :ref:`format specification <formatspec>` to present the value.
+   :type format_spec: str
+
+   The :class:`!Interpolation` type represents an expression inside a template string.
+
+   :class:`!Interpolation` instances are shallow immutable: their attributes cannot be
+   reassigned.
+
+   .. attribute:: value
+
+       :returns: The evaluated value of the interpolation.
+       :rtype: object
```

Comment:
If it's an attribute, use `:type:` role instead of rtype (https://www.sphinx-doc.org/en/master/usage/domains/python.html#directive-py-attribute).